tracks/julia/exercises/nucleotide-count/mentoring.md

```diff
@@ -0,0 +1,40 @@
+### Mentor guidance
+
+Students that use four counting variables rather than mutating a dict should be shown the dict solution or asked to come up with it on their own if there is some doubt that they knew they could do that.
+
+We shouldn't require the mutating-a-dict strategy because it's actually pretty slow.
+
+
+### Example solutions
+
+```julia
+function count_nucleotides(strand)
+    counter = Dict(base => 0 for base in "ACGT")
+    for base in strand
+        if haskey(counter, base)
+            counter[base] += 1
+        else
+            throw(DomainError(base, "only A, C, G and T are valid nucleotides"))
+        end
+    end
+    return counter
+end
+```
+
+```julia
+function count_nucleotides(strand)
+    counts = Dict(count(==(base), strand) for base in "ACGT")
+    if sum(values(counts)) != length(strand)
```

Comment:
This also - for Julia's `String` type, because it is UTF-8 encoded (loosely speaking, it's not validated, which makes it even slower), `length(strand)` is an O(N) operation - so if you are checking a lot of strands / long strands, that will be a lot slower than it needs to be.

Reply:
I know, but I don't think that makes this a bad solution.

Reply:
Well - anything that deals with DNA samples, and doesn't scale, I don't think would be a useful solution for any real-world problem.